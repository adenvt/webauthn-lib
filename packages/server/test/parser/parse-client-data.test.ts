import attestationResponse from '../__sample__/register.json'
import assertionResponse from '../__sample__/login.json'
import parseClientData from '../../src/parser/parse-client-data'

test('Parsing Attestation Client Data', () => {
  const clientData = parseClientData(attestationResponse.response.clientDataJSON)

  expect(clientData).toHaveProperty('challenge')
  expect(clientData).toHaveProperty('origin')
  expect(clientData).toHaveProperty('type')

  expect(clientData.type).toBe('webauthn.create')
  expect(clientData).toMatchSnapshot()
})

test('Parsing Assertion Client Data', () => {
  const clientData = parseClientData(assertionResponse.response.clientDataJSON)

  expect(clientData).toHaveProperty('challenge')
  expect(clientData).toHaveProperty('origin')
  expect(clientData).toHaveProperty('type')

  expect(clientData.type).toBe('webauthn.get')
  expect(clientData).toMatchSnapshot()
})
