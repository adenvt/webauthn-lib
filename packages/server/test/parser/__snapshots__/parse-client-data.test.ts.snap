// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parsing Assertion Client Data 1`] = `
Object {
  "challenge": "fgqnwaiAcwNtHB48dSdU__sk7VNSo41fkPHGuM-W_pI",
  "origin": "https://webauthn.io",
  "type": "webauthn.get",
}
`;

exports[`Parsing Attestation Client Data 1`] = `
Object {
  "challenge": "7LhOEoxKASIyXaOY7_cGDxnE3PDD4pB4elK7PVh9s2Q",
  "origin": "https://webauthn.io",
  "type": "webauthn.create",
}
`;
